docs/sources/flow/get-started/install/puppet.md

```diff
@@ -0,0 +1,105 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/get-started/install/puppet/
+canonical: https://www.puppet.com/
+description: Learn how to install Grafana Agent Flow with Puppet
+menuTitle: Puppet
+title: Install Grafana Agent Flow with Puppet
+weight: 550
+---
+
+# Install {{% param "PRODUCT_NAME" %}} with Puppet
+
+You can use Puppet to install and manage {{< param "PRODUCT_NAME" >}}.
+
+## Before you begin
+
+- These steps assume you already have a working [Puppet][] setup.
+- You can add the following manifest to any new or existing module you choose.
+- The manifest installs {{< param "PRODUCT_NAME" >}} from the package repositories. The manifest targets Linux systems from the following families:
+  - Debian (including Ubuntu)
+  - RedHat Enterprise Linux (including Fedora)
+
+## Steps
+
+To add {{% param "PRODUCT_NAME" %}} to a host:
+
+1. Ensure that the following module dependencies are declared and installed:
+    ```json
+    {
+    "name": "puppetlabs/apt",
+    "version_requirement": ">= 4.1.0 <= 7.0.0"
+    },
+    {
+    "name": "puppetlabs/yumrepo_core",
+    "version_requirement": "<= 2.0.0"
+    }
+    ```
+1. Create a new [Puppet][] manifest with the following resources to add the Grafana package repositories, install the `grafana-agent-flow` package, and run the service:
+    ```ruby
+    class grafana_agent::grafana_agent_flow () {
+      case $::os['family'] {
+        'debian': {
+          apt::source { 'grafana':
+            location => 'https://apt.grafana.com/',
+            release  => '',
+            repos    => 'stable main',
+            key      => {
+              id     => 'B53AE77BADB630A683046005963FA27710458545',
+              source => 'https://apt.grafana.com/gpg.key',
+            },
+          } -> package { 'grafana-agent-flow':
+            require => Exec['apt_update'],
+          } -> service { 'grafana-agent-flow':
+            ensure    => running,
+            name      => 'grafana-agent-flow',
+            enable    => true,
+            subscribe => Package['grafana-agent-flow'],
+          }
+        }
+        'redhat': {
+          yumrepo { 'grafana':
+            ensure   => 'present',
+            name     => 'grafana',
+            descr    => 'grafana',
+            baseurl  => 'https://packages.grafana.com/oss/rpm',
+            gpgkey   => 'https://packages.grafana.com/gpg.key',
+            enabled  => '1',
+            gpgcheck => '1',
+            target   => '/etc/yum.repo.d/grafana.repo',
+          } -> package { 'grafana-agent-flow':
+          } -> service { 'grafana-agent-flow':
+            ensure    => running,
+            name      => 'grafana-agent-flow',
+            enable    => true,
+            subscribe => Package['grafana-agent-flow'],
+          }
+        }
+        default: {
+          fail("Unsupported OS family: (${$::os['family']})")
+        }
+      }
+    }
+    ```
+1. To use this manifest, add the following line to your desired manifests:
```

Comment:
It's been a while since I did Puppet, but this sounds a bit odd. Wouldn't the file above be called something like a module and then in the manifest you include a module? I couldn't find clear definitions in https://www.puppet.com/docs/puppet/8/dirs_important_directories.